modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/InteractListener.java

```diff
@@ -99,30 +118,24 @@ public void onPlayerInteractStick(PlayerInteractEvent event) {
                 int DY = 1;
                 if (event.getPlayer().isSneaking())
                     DY = -1;
-
+                DY *= craft.getCurrentGear();
                 craft.translate(0, DY, 0);
                 timeMap.put(event.getPlayer(), System.currentTimeMillis());
                 craft.setLastCruiseUpdate(System.currentTimeMillis());
                 return;
             }
             // Player is onboard craft and right clicking
-            float rotation = (float) Math.PI * event.getPlayer().getLocation().getYaw() / 180f;
-
-            float nx = -(float) Math.sin(rotation);
-            float nz = (float) Math.cos(rotation);
-
-            int dx = (Math.abs(nx) >= 0.5 ? 1 : 0) * (int) Math.signum(nx);
-            int dz = (Math.abs(nz) > 0.5 ? 1 : 0) * (int) Math.signum(nz);
-            int dy;
 
+            final Vector direction = player.getLocation().getDirection();
             float p = event.getPlayer().getLocation().getPitch();
 
-            dy = -(Math.abs(p) >= 25 ? 1 : 0) * (int) Math.signum(p);
+            direction.setY(-(Math.abs(p) >= 25 ? 1 : 0) * (int) Math.signum(p));
+            direction.normalize();
+            direction.multiply(currentGear);
 
-            if (Math.abs(event.getPlayer().getLocation().getPitch()) >= 75) {
-                dx = 0;
-                dz = 0;
-            }
+            int dx = (int) rint(direction.getX());
+            int dz = (int) rint(direction.getZ());
+            int dy = (int) rint(direction.getY());
```

Comment:
Has this code change been confirmed to not change functionality?  CC thought it did but I see no changes to resolve his issue.

Reply:
This is probably the cause of #356 .  @eirikh1996 in the future please do not change code in PRs if it is not needed.